### PR TITLE
Fix e2e certificates generation with OpenSSL 3

### PR DESCRIPTION
**Pull request by jsoriano**
## What is the problem this PR solves?

Fix private key parsing issue with certificates created for e2e tests.
```
tls: failed to parse private key
```

## How does this PR solve the problem?

OpensSSL 3 uses PKCS#8 by default instead of PKCS#1. This changes the format of encrypted private keys, fleet server is not able to parse encrypted PKCS#8 keys (see https://github.com/elastic/elastic-agent-libs/issues/134).

This PR forces OpenSSL 3 to use the old format when creating keys, to ensure that fleet-server is able to read them.

Longer term we should fix https://github.com/elastic/elastic-agent-libs/issues/134 to ensure that private keys generated with more recent libraries can be used.

## Design Checklist

- [ ] I have ensured my design is stateless and will work when multiple fleet-server instances are behind a load balancer.
- [ ] I have or intend to scale test my changes, ensuring it will work reliably with 100K+ agents connected.
- [ ] I have included fail safe mechanisms to limit the load on fleet-server: rate limiting, circuit breakers, caching, load shedding, etc.

## Related issues

- Root cause: https://github.com/elastic/elastic-agent-libs/issues/134

### CODE REVIEW

File: dev-tools/e2e/validatecerts.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package main
```
Comment (michel-laterman) [inline dev-tools/e2e/validatecerts.go:5]:
```suggestion

// validatecerts is a script to ensure that certs used in e2e tests can be parsed by tlscommon
package main
```
(nit)

**Review by michel-laterman — APPROVED** [review]
lgtm

**Comment by mrodm** [comment]
buildkite test this

**Pull request merged**